src/components/datepicker/DatePickerInput.js

```diff
@@ -141,6 +141,7 @@ class DatePickerInput extends PureComponent {
           position="end"
           offsetCorrection={30}
           zIndex={500}
+          returnFocusToSource={false}
```

Comment:
As your commit message says, the datepicker input is a bit weird right now. You actually can't get past the datepicke input right now, because when it closes, it goes back to the body.

Reply:
Yup, most of it is caused by the date picker opening when focused, and the datepicker doesn't pass correct events to its handlers. I think a separate focus state for the date picker input without opening the popover and a secondary action like f.e. pressing enter could solve this.

Reply:
Planned a meeting to discuss this, in the meantime I don't think this is blocking for this pull request.